tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ImportLiferayModuleProjectOp.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.modules;
+
+import org.eclipse.sapphire.ElementType;
+import org.eclipse.sapphire.ExecutableElement;
+import org.eclipse.sapphire.Value;
+import org.eclipse.sapphire.ValueProperty;
+import org.eclipse.sapphire.modeling.Path;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.modeling.annotations.AbsolutePath;
+import org.eclipse.sapphire.modeling.annotations.DelegateImplementation;
+import org.eclipse.sapphire.modeling.annotations.FileSystemResourceType;
+import org.eclipse.sapphire.modeling.annotations.Label;
+import org.eclipse.sapphire.modeling.annotations.Required;
+import org.eclipse.sapphire.modeling.annotations.Service;
+import org.eclipse.sapphire.modeling.annotations.Type;
+import org.eclipse.sapphire.modeling.annotations.ValidFileSystemResourceType;
+
+/**
+ * @author Andy Wu
+ */
+public interface ImportLiferayModuleProjectOp extends ExecutableElement
+{
+
+    ElementType TYPE = new ElementType( ImportLiferayModuleProjectOp.class );
+
+    // *** Location ***
+
+    @Label( standard = "location" )
+    @Type( base = Path.class )
+    @AbsolutePath
+    @ValidFileSystemResourceType( FileSystemResourceType.FOLDER )
+    @Service( impl = ImportModulsProjectLocationValidationService.class )
+    @Required
+    ValueProperty PROP_LOCATION = new ValueProperty( TYPE, "Location" );
+
+    Value<Path> getLocation();
+
+    void setLocation( String value );
+
+    void setLocation( Path value );
```

Comment:
We should have a new read only property that is Derived that will be the "build type" just so the user can see what build type we have detected (either gradle or maven for now).